build/_build/Build.Gitlab.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Amazon.SimpleSystemsManagement;
+using Amazon.SimpleSystemsManagement.Model;
+using Nuke.Common;
+using Nuke.Common.IO;
+using Nuke.Common.Tools.NuGet;
+using Nuke.Common.Tools.SignTool;
+using Nuke.Common.Utilities.Collections;
+using static Nuke.Common.EnvironmentInfo;
+using Target = Nuke.Common.Target;
+
+partial class Build
+{
+    // Only change temporarily for testing
+    bool UseTestPfxCertificate = false;
+
+    Target SignPackages => _ => _
+        .Description("Sign the dlls, NupKg and msi files after build")
+        .Unlisted()
+        .Requires(() => IsWin)
+        .After(BuildTracerHome, PackageTracerHome)
+        .Executes(async () =>
+        {
+            // To create a pfx certificate for local testing, use powershell and run:
+            // $outputLocation = "test_cert.pfx"
+            // $cert = New-SelfSignedCertificate -DnsName sample.contoso.com -Type CodeSigning -CertStoreLocation Cert:\CurrentUser\My
+            // $CertPassword = ConvertTo-SecureString -String "Passw0rd" -Force –AsPlainText
+            // Export-PfxCertificate -Cert "cert:\CurrentUser\My\$($cert.Thumbprint)" -FilePath $outputLocation -Password $CertPassword
+
+            var tempFileName = Path.GetTempFileName();
+            const string timestampServer = "http://timestamp.digicert.com/";
+
+            try
+            {
+                var (certPath, certPassword) = UseTestPfxCertificate
+                    ? (@"test_cert.pfx", "Passw0rd")
+                    : await GetSigningMaterial(tempFileName);
+
+                Logger.Info("Signing material retrieved, signing binaries...");
+                TracerHomeDirectory.GlobFiles("**/Datadog*.dll")
+                    .Concat(ArtifactsDirectory.GlobFiles("**/*.msi"))
+                    .ForEach(binaryPath => SignBinary(certPath, certPassword, binaryPath));
```

Comment:
It looks like this step signs both dll and msi files, which means the msi was already built with unsigned dlls. Is that correct? What we want to do is:
1. build dlls
2. sign dlls
3. build msi (using the signed dlls from step 2)
4. build nuget (using the signed dlls from step 2)

Reply:
Good catch. I was tricked by undescriptive msbuild target names 😅 

I thought this built the MSI because it defines MSI variables:
https://github.com/DataDog/dd-trace-dotnet/blob/84d1f1b9d8fff82eb2c4e7aba88c5b357ad35195/tasks/dotnettracer.py#L41

Turns out we don't build it until here (after signing dlls) 
https://github.com/DataDog/dd-trace-dotnet/blob/84d1f1b9d8fff82eb2c4e7aba88c5b357ad35195/tasks/dotnettracer.py#L67-L75

I'll update the PR 👍

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1512/commits/f3b5703dc9731b3b51c55be11c5cd2bfcff90dd9

Sequence looks better now:

```
════════════════════════════════════════════════════
Target                          Status      Duration
────────────────────────────────────────────────────
Restore                         Executed        0:16
CreateRequiredDirectories       Executed      < 1sec
CompileManagedSrc               Executed        0:19
PublishManagedProfiler          Executed        0:06
CompileNativeSrcMacOs           Skipped                // EnvironmentInfo.IsOsx
CompileNativeSrcLinux           Skipped                // EnvironmentInfo.IsLinux
CompileNativeSrcWindows         Executed        0:02
PublishNativeProfilerMacOs      Skipped                // EnvironmentInfo.IsOsx
PublishNativeProfilerLinux      Skipped                // EnvironmentInfo.IsLinux
PublishNativeProfilerWindows    Executed      < 1sec
CopyIntegrationsJson            Executed      < 1sec
SignDlls                        Executed        0:13
ZipTracerHome                   Executed        0:02
BuildMsi                        Executed        0:15
PackNuGet                       Executed        0:02
SignMsiAndNupkg                 Executed        0:03
────────────────────────────────────────────────────
Total                                           1:21
════════════════════════════════════════════════════
```